spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/config/EnvironmentRepositoryConfiguration.java

```diff
@@ -301,6 +317,19 @@ public NativeEnvironmentRepository nativeEnvironmentRepository(
 
 }
 
+@Configuration
+@Profile("s3")
+class AwsS3RepositoryConfiguration {
+
+	@Bean
+	public AwsS3EnvironmentRepository awsS3EnvironmentRepository(
```

Comment:
Probably should make this conditional to be safe

Reply:
Conditional on classpath, profile something else? Getting the wiring correct on the new S3 config was one of the parts I was least confident that I did correctly.

Reply:
`@ConditionalOnMissingBean`

Reply:
fixed, except I used @ConditionalOnClass(AmazonS3Client.class) and @ConditionalOnBean(AmazonS3Client.class) like the other RepositoryConfigurations.

Reply:
I dont see why it is conditional on `AmazonS3Client` it doesn't use it
Also the reason I was suggesting adding `@ConditionalOnMissingBean(AwsS3EnvironmentRepository)` is so someone could optionally provide their own if they needed to tweak anything